coffee/scheduler.py

```diff
@@ -835,7 +836,7 @@ def reschedule(self, root):
             # Split the main expressions to maximize the impact of the rescheduling (this
             # helps if different summands have zero-valued blocks at different offsets)
             elf = ExpressionFissioner(cut=1, loops='none')
-            for stmt, expr_info in self.exprs.items():
+            for stmt, expr_info in list(iteritems(self.exprs)):
                 if expr_info.is_scalar:
                     continue
                 self.exprs.pop(stmt)
```

Comment:
it's fun because the more I look at it, and the less I understand. Bho. Sometimes I do things and I simply don't know why I do them that way.